test/contlinear.jl

```diff
@@ -726,6 +750,100 @@ function linear7test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
     end
 end
 
+function linear8test(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+    @testset "test infeasible problem" begin
+        # min x
+        # s.t. 2x+y <= -1
+        # x,y >= 0
+        @test MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64}, [(MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}),(MOI.SingleVariable,MOI.GreaterThan{Float64})])
+
+        m = MOI.SolverInstance(solver)
+        x = MOI.addvariable!(m)
+        y = MOI.addvariable!(m)
+        c = MOI.addconstraint!(m, MOI.ScalarAffineFunction([x,y], [2.0,1.0], 0.0), MOI.LessThan(-1.0))
+        MOI.addconstraint!(m, MOI.SingleVariable(x), MOI.GreaterThan(0.0))
+        MOI.addconstraint!(m, MOI.SingleVariable(y), MOI.GreaterThan(0.0))
+        MOI.setobjective!(m, MOI.MinSense, MOI.ScalarAffineFunction([x], [1.0], 0.0))
+        MOI.optimize!(m)
+
+        @test MOI.cangetattribute(m, MOI.ResultCount())
+        if MOI.getattribute(m, MOI.ResultCount()) == 1
+            # solver returned an infeasibility ray
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+            @test MOI.getattribute(m, MOI.DualStatus()) == MOI.InfeasibilityCertificate
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), c)
+            @test MOI.getattribute(m, MOI.ConstraintDual(), c) ≈ -1 atol=ε
+        else
+            # solver returned nothing
+            @test MOI.getattribute(m, MOI.ResultCount()) == 0
+            @test MOI.cangetattribute(m, MOI.PrimalStatus(1)) == false
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.InfeasibleNoResult ||
+                MOI.getattribute(m, MOI.TerminationStatus()) == MOI.InfeasibleOrUnbounded
+        end
+    end
+    @testset "test unbounded problem" begin
+        # min -x-y
+        # s.t. -x+2y <= 0
+        # x,y >= 0
+        @test MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64}, [(MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}),(MOI.SingleVariable,MOI.GreaterThan{Float64})])
+
+        m = MOI.SolverInstance(solver)
+        x = MOI.addvariable!(m)
+        y = MOI.addvariable!(m)
+        MOI.addconstraint!(m, MOI.ScalarAffineFunction([x,y], [-1.0,2.0], 0.0), MOI.LessThan(0.0))
+        MOI.addconstraint!(m, MOI.SingleVariable(x), MOI.GreaterThan(0.0))
+        MOI.addconstraint!(m, MOI.SingleVariable(y), MOI.GreaterThan(0.0))
+        MOI.setobjective!(m, MOI.MinSense, MOI.ScalarAffineFunction([x, y], [-1.0, -1.0], 0.0))
+        MOI.optimize!(m)
+
+        @test MOI.cangetattribute(m, MOI.ResultCount())
+        if MOI.getattribute(m, MOI.ResultCount()) == 1
+            # solver returned an unbounded ray
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.ReductionCertificate
```

Comment:
Actually the ray status is there and is called `InfeasibilityCertificate` because a primal unbounded ray is an infeasibility certificate of the dual. Facial reduction certificates are mentioned briefly [in the manual](http://www.juliaopt.org/MathOptInterface.jl/latest/apimanual.html#Primal-dual-convex-solver-1); they're for cases of duality failures or unattained solutions which never happen for LP.